packages/ui/src/views/agentflows/index.jsx

```diff
@@ -1,233 +1,243 @@
+        setSearch(event.target.value)
     }
 
-    const goToCanvas = (selectedAgentflow) => {
-        navigate(`/agentcanvas/${selectedAgentflow.id}`)
+    const handleCategoryChange = (event) => {
+        setCategoryFilter(event.target.value)
     }
 
     useEffect(() => {
-        getAllAgentflows.request()
-
-        // eslint-disable-next-line react-hooks/exhaustive-deps
+        getAllAgentflowsApi.request()
+        getMarketplaceAgentflowsApi.request()
     }, [])
 
     useEffect(() => {
-        if (getAllAgentflows.error) {
-            if (getAllAgentflows.error?.response?.status === 401) {
-                setLoginDialogProps({
-                    title: 'Login',
-                    confirmButtonName: 'Login'
-                })
-                setLoginDialogOpen(true)
-            } else {
-                setError(getAllAgentflows.error)
-            }
+        if (getAllAgentflowsApi.error || getMarketplaceAgentflowsApi.error) {
+            setError(getAllAgentflowsApi.error || getMarketplaceAgentflowsApi.error)
         }
-    }, [getAllAgentflows.error])
+    }, [getAllAgentflowsApi.error, getMarketplaceAgentflowsApi.error])
 
     useEffect(() => {
-        setLoading(getAllAgentflows.loading)
-    }, [getAllAgentflows.loading])
+        setLoading(getAllAgentflowsApi.loading || getMarketplaceAgentflowsApi.loading)
+    }, [getAllAgentflowsApi.loading, getMarketplaceAgentflowsApi.loading])
 
     useEffect(() => {
-        if (getAllAgentflows.data) {
-            try {
-                const agentflows = getAllAgentflows.data
-                const images = {}
-                for (let i = 0; i < agentflows.length; i += 1) {
-                    const flowDataStr = agentflows[i].flowData
-                    const flowData = JSON.parse(flowDataStr)
-                    const nodes = flowData.nodes || []
-                    images[agentflows[i].id] = []
-                    for (let j = 0; j < nodes.length; j += 1) {
-                        const imageSrc = `${baseURL}/api/v1/node-icon/${nodes[j].data.name}`
-                        if (!images[agentflows[i].id].includes(imageSrc)) {
-                            images[agentflows[i].id].push(imageSrc)
-                        }
+        if (getAllAgentflowsApi.data && getMarketplaceAgentflowsApi.data) {
+            const processFlowData = (flows) => {
+                const processedImages = {}
+                const processedNodeTypes = {}
+                flows.forEach((flow) => {
+                    if (flow && flow.flowData) {
+                        const flowData = JSON.parse(flow.flowData)
+                        const nodes = flowData.nodes || []
+                        processedImages[flow.id] = []
+                        processedNodeTypes[flow.id] = []
+                        nodes.forEach((node) => {
+                            if (['Multi Agents', 'Chat Models', 'Tools', 'Document Loaders'].includes(node.data.category)) {
```

Comment:
Should check that node.data exists at least.   I'd recommended doing the same for the other items below as well